src/main/java/org/opensearch/security/resources/ResourceAccessControlTransportClient.java

```diff
@@ -0,0 +1,257 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.security.resources;
+
+import java.util.Set;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import org.opensearch.OpenSearchStatusException;
+import org.opensearch.Version;
+import org.opensearch.cluster.service.ClusterService;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.core.action.ActionListener;
+import org.opensearch.core.rest.RestStatus;
+import org.opensearch.security.resources.rest.list.ListAccessibleResourcesAction;
+import org.opensearch.security.resources.rest.list.ListAccessibleResourcesRequest;
+import org.opensearch.security.resources.rest.revoke.RevokeResourceAccessAction;
+import org.opensearch.security.resources.rest.revoke.RevokeResourceAccessRequest;
+import org.opensearch.security.resources.rest.revoke.RevokeResourceAccessResponse;
+import org.opensearch.security.resources.rest.share.ShareResourceAction;
+import org.opensearch.security.resources.rest.share.ShareResourceRequest;
+import org.opensearch.security.resources.rest.share.ShareResourceResponse;
+import org.opensearch.security.resources.rest.verify.VerifyResourceAccessAction;
+import org.opensearch.security.resources.rest.verify.VerifyResourceAccessRequest;
+import org.opensearch.security.resources.rest.verify.VerifyResourceAccessResponse;
+import org.opensearch.security.spi.resources.ShareableResource;
+import org.opensearch.security.spi.resources.client.ResourceSharingClient;
+import org.opensearch.security.spi.resources.sharing.ResourceSharing;
+import org.opensearch.security.spi.resources.sharing.SharedWithActionGroup;
+import org.opensearch.security.support.ConfigConstants;
+import org.opensearch.transport.client.Client;
+
+import static org.opensearch.security.resources.ResourceSharingConstants.RESOURCE_SHARING_MIN_SUPPORTED_VERSION;
+
+/**
+ * Access control transport client responsible for handling resource sharing operations such as verifying,
+ * sharing, revoking, and listing access to shareable resources.
+ * NOTE: The calls are gated behind `cluster:admin/security/resource_access/*` action permissions
+ *
+ * @opensearch.experimental
+ */
+public final class ResourceAccessControlTransportClient implements ResourceSharingClient {
```

Comment:
This class is not used and is added for future purposes in-case we decide to gate the transport calls for resource access-control.

Reply:
IMHO, we should not add unused code to the code base. I'd recommend to keep it in a branch and add it when it is actually needed.

Reply:
removed.